advanced/calculator/calc_with_gui.py

```diff
@@ -0,0 +1,208 @@
+        else:
+            input_box.insert(text_len, value)
+
+    def is_this_solution(self, string_calc):
+        for char in string_calc[1:]:
+            if self.is_char_op(char):
+                return False
+        return True
+
+    @staticmethod
+    def is_char_op(char):
+        op_lst = ["+", "-", "*", "/", "^"]
+        if char in op_lst:
+            return True
+        return False
+
+    def pulling_the_first_calc(self, string_calc, strong_op_index):
+        flag = False
+        first_index = strong_op_index
+        last_index = strong_op_index
+
+        while not flag:
+            first_index -= 1
+            if first_index == 0:
+                flag = True
+            elif self.is_char_op(string_calc[first_index]):
+                first_index += 1
+                flag = True
+
+        flag = False
+        while not flag:
+            last_index += 1
+            if last_index == len(string_calc) - 1:
+                last_index += 1
+                flag = True
+            elif self.is_char_op(string_calc[last_index]) and not last_index == strong_op_index + 1:
+                flag = True
+
+        return string_calc[:first_index], string_calc[first_index:last_index], string_calc[last_index:]
+
+    @staticmethod
+    def index_strongest(string_calc):
+
+        if "^" in string_calc:
+            return string_calc.index("^")
+
+        if "*" in string_calc or "/" in string_calc:
+            try:
+                multi_index = string_calc.index("*")
+            except ValueError:
+                multi_index = len(string_calc)
+            try:
+                divide_index = string_calc.index("/")
+            except ValueError:
+                divide_index = len(string_calc)
+
+            return min(multi_index, divide_index)
+
+        if "-" in string_calc or "+" in string_calc[1:]:
+            try:
+                plus_index = string_calc[1:].index("+") + 1
+            except ValueError:
+                plus_index = len(string_calc)
+            try:
+                minus_index = string_calc[1:].index("-") + 1
+            except ValueError:
+                minus_index = len(string_calc)
+
+            return min(plus_index, minus_index)
+    """ 
+    def get_answer(self, string_calc, input_box):
+        list_calc = self.organize_input(string_calc)
+        solution = self.simple_calc(list_calc)
+        self.configure_entry_text(solution, input_box)
+        print(str(solution))
+    """
+
+    def get_answer(self, string_calc, input_box):
+        if self.is_this_solution(string_calc):
+            return self.configure_entry_text(float(string_calc), input_box)
+
+        strong_op_index = self.index_strongest(string_calc)
+        string_calc_tuple = self.pulling_the_first_calc(string_calc, strong_op_index)
+        list_calc = self.organize_input(string_calc_tuple[1])
+        return self.get_answer(string_calc_tuple[0] + str(self.simple_calc(list_calc)) + string_calc_tuple[2],
```

Comment:
I didn't quite got your solution, its complicated.
think of the recursion like this: you get an expression, and you have found the strongest operator.
now, you have (expression)(operator)(expression), and you call the recursive method for each expression.
this solution is much more simpler and readable.